packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -355,12 +355,19 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
     } else if (event is PointerUpEvent || event is PointerCancelEvent) {
       hitTestResult = _hitTests.remove(event.pointer);
     } else if (event.down) {
-      // Because events that occur with the pointer down (like
-      // [PointerMoveEvent]s) should be dispatched to the same place that their
-      // initial PointerDownEvent was, we want to re-use the path we found when
-      // the pointer went down, rather than do hit detection each time we get
-      // such an event.
-      hitTestResult = _hitTests[event.pointer];
+      if (event.kind == PointerDeviceKind.mouse) {
```

Comment:
Actually, I want to replace all the logic with this, but I can't fully evaluate the impact, so I limit the changes to mouse operations.